lib/floe/workflow/runner/docker.rb

```diff
@@ -60,6 +63,62 @@ def output(runner_context)
           runner_context["output"] = output
         end
 
+        def wait(contexts = [], timeout: nil, since: Time.now.utc, events: [], &block)
+          refs              = Array(contexts).map { |ctx| ctx["container_ref"] }
+          container_filters = refs.map   { |container_id| [:filter, "container=#{container_id}"] } << [:filter, "type=container"]
+          event_filters     = events.map { |event| [:filter, "event=#{event}"] }
+
+          until_timestamp = Time.now.utc + timeout if timeout
+
+          r, w = IO.pipe
+
+          params = ["events", [:format, "{{json .}}"]]
+          params += event_filters                  if event_filters.any?
+          params += container_filters              if container_filters.any?
```

Comment:
this will always have a value

Reply:
true, had this before I added the type=container filter

Reply:
pulling out a new method `build_params` may trim down `wait` enough to make code climate happier?

It would contain  (refs, container_filter, event_filter, params)